autosklearn/automl.py

```diff
@@ -917,6 +936,168 @@ def refit(self, X, y):
+        self,
+        X: np.ndarray,
+        y: np.ndarray,
+        task: int,
+        is_classification: bool,
+        dataset_name: Optional[str] = None,
+        X_test: Optional[np.ndarray] = None,
+        y_test: Optional[np.ndarray] = None,
+        **kwargs: Dict,
+    ) -> Tuple[Optional[BasePipeline], RunInfo, RunValue]:
+        """ Fits and individual pipeline configuration and returns
+        the result to the user.
+
+        The Estimator constraints are honored, for example the resampling
+        strategy, or memory constraints, unless directly provided to the method.
+        By default, this method supports the same signature as fit(), and any extra
+        arguments are redirected to the TAE evaluation function, which allows for
+        further customization while building a pipeline.
+
+        Parameters
+        ----------
+        X: array-like, shape = (n_samples, n_features)
+            The features used for training
+        y: array-like
+            The labels used for training
+        X_test: Optionalarray-like, shape = (n_samples, n_features)
+            If provided, the testing performance will be tracked on this features.
+        y_test: array-like
+            If provided, the testing performance will be tracked on this labels
+        config: Configuration
+            A configuration object used to define a pipeline steps
+        dataset_name: Optional[str]
+            A string to tag and identify the Auto-Sklearn run
+        is_classification: bool
+            Whether the task is for classification or regression. This affects
+            how the targets are treated
+
+        Returns
+        -------
+        pipeline: Optional[BasePipeline]
+            The fitted pipeline. In case of failure while fitting the pipeline,
+            a None is returned.
+        run_info: RunInFo
+            A named tuple that contains the configuration launched
+        run_value: RunValue
+            A named tuple that contains the result of the run
+        """
+
+        # Get the configuration space
+        # This also ensures that the Backend has processed the
+        # dataset
+        if self.configuration_space is None:
+            self.configuration_space = self.fit(
+                X=X, y=y, task=task,
+                dataset_name=dataset_name,
+                X_test=kwargs.pop('X_test', None),
+                y_test=kwargs.pop('y_test', None),
+                feat_type=kwargs.pop('feat_type', None),
+                only_return_configuration_space=True)
+
+        # Get a configuration from the user or sample from the CS
+        config = kwargs.pop('config', self.configuration_space.sample_configuration())
+
+        # We do not want to overwrite existing runs
+        self.num_run += 1
+        config.config_id = self.num_run
+
+        exclude = dict()
+        include = dict()
+        if self._include_preprocessors is not None and self._exclude_preprocessors is not None:
```

Comment:
Can this and the functionality in `fit` maybe be extracted to a separate function?

Reply:
Thanks, I have done so